src/encoded/static/components/view_controls.js

```diff
@@ -0,0 +1,325 @@
+            <p>
+                Click the &ldquo;Download&rdquo; button below to download a &ldquo;files.txt&rdquo; file that contains a list of URLs to a file containing all the experimental metadata and links to download the file.
+                The first line of the file has the URL or command line to download the metadata file.
+            </p>
+            <p>
+                Further description of the contents of the metadata file are described in the <a href="/help/batch-download/">Batch Download help doc</a>.
+            </p>
+            <p>
+                The &ldquo;files.txt&rdquo; file can be copied to any server.<br />
+                The following command using cURL can be used to download all the files in the list:
+            </p>
+            <code>xargs -L 1 curl -O -L &lt; files.txt</code><br />
+            <div>{additionalContent}</div>
+        </ModalBody>
+        <ModalFooter
+            closeModal={<button className="btn btn-default">Close</button>}
+            submitBtn={<button id="batch-download-submit" className="btn btn-info" disabled={disabled} onClick={handleDownloadClick}>Download</button>}
+            dontClose
+        />
+    </Modal>
+);
+
+BatchDownloadModal.propTypes = {
+    /** Function to call when Download button gets clicked */
+    handleDownloadClick: PropTypes.func.isRequired,
+    /** Title to override usual actuator "Download" button title */
+    title: PropTypes.string,
+    /** True to disable Download button */
+    disabled: PropTypes.bool,
+    /** Additional content in modal as component */
+    additionalContent: PropTypes.object,
+};
+
+BatchDownloadModal.defaultProps = {
+    title: 'Download',
+    disabled: false,
+    additionalContent: null,
+};
+
+
+/**
+ * Display batch download button if the search results qualify for one.
+ */
+export class BatchDownloadControls extends React.Component {
+    constructor(props) {
+        super(props);
+        this.handleDownloadClick = this.handleDownloadClick.bind(this);
+    }
+
+    handleDownloadClick() {
+        const queryString = getQueryFromFilters(this.props.results.filters);
+        this.context.navigate(`/batch_download/${queryString}`);
+    }
+
+    render() {
+        const { results } = this.props;
+
+        // No Download button if the search path is prohibited.
+        const hasProhibitedPath = BATCH_DOWNLOAD_PROHIBITED_PATHS.some(path => results['@id'].startsWith(path));
+        if (hasProhibitedPath) {
+            return null;
+        }
+
+        // No download button if "type=" for an allowed type doesn't exist in query string.
+        const hasDownloadDocType = results.filters.some(filter => filter.field === 'type' && BATCH_DOWNLOAD_DOC_TYPES.includes(filter.term));
+        if (!hasDownloadDocType) {
+            return null;
+        }
+
+        // No download button if no files.
+        const hasFiles = results.facets.some(facet => facet.field === 'files.file_type' && facet.total > 0);
```

Comment:
(Minor). Maybe facet.total should be first?

Reply:
Would that be better? The hope being the first test fails more than the second so that the test would short circuit, and I think `facet.field === 'files.file_type'` would fail more than `facet.total > 0`.